ldm/models/diffusion/shared_invokeai_diffusion.py

```diff
@@ -275,17 +281,15 @@ def _combine(self, unconditioned_next_x, conditioned_next_x, guidance_scale):
         combined_next_x = unconditioned_next_x + scaled_delta
         return combined_next_x
 
-    def _threshold(self, threshold, warmup, latents: torch.Tensor, sigma) -> torch.Tensor:
-        warmup_scale = (1 - sigma.item() / 1000) / warmup if warmup else math.inf
-        if warmup_scale < 1:
-            # This arithmetic based on https://github.com/invoke-ai/InvokeAI/pull/395
-            warming_threshold = 1 + (threshold - 1) * warmup_scale
-            current_threshold = np.clip(warming_threshold, 1, threshold)
+    def _threshold(self, threshold, warmup, latents: torch.Tensor, percent_through) -> torch.Tensor:
+        if percent_through < warmup:
+            current_threshold = threshold + threshold * (1 - (percent_through / warmup))
```

Comment:
this abandons the previous limit where current_threshold can't go below 1, was that intentional?

Reply:
What does a negative threshold mean?